Framework/Core/include/Framework/TableTreeHelpers.h

```diff
@@ -22,122 +22,372 @@ namespace o2::framework
-//  . TableToTree t2t(ta,f,treename);
-//  . t2t.addBranch(coumn1); t2t.addBranch(coumn1); ...
-//    OR
-//    t2t.addAllBranches();
-//  . t2t.process();
+//  . TableToTree t2t(f,treename);
+//  . t2t.addBranches(ta);
+//    OR t2t.addBranch(column.get(), field.get()), ...;
+//  . t2t.write();
 //
 // .............................................................................
-class BranchIterator
+class ColumnToBranchBase
 {
+ public:
+  ColumnToBranchBase(arrow::ChunkedArray* column, arrow::Field* field, int size = 1);
+  virtual ~ColumnToBranchBase(){};
+  void at(int64_t* pos)
+  {
+    mCurrentPos = pos;
+    resetBuffer();
+  }
 
- private:
-  std::string mBranchName;    // branch name
-  arrow::ArrayVector mChunks; // chunks
-  Int_t mNumberChuncs;        // number of chunks
-  Int_t mCounterChunk;        // chunk counter
-  Int_t mNumberRows;          // number of rows
-  Int_t mCounterRow;          // row counter
-
-  // data buffers for each data type
-  bool mStatus = false;
-  arrow::Field* mField;
-  arrow::Type::type mFieldType;
-  arrow::Type::type mElementType;
-  int32_t mNumberElements;
-  std::string mLeaflistString;
-
-  TBranch* mBranchPtr = nullptr;
-
-  char* mBranchBuffer = nullptr;
-  void* mValueBuffer = nullptr;
-
-  std::shared_ptr<arrow::BooleanArray> mArray_o = nullptr;
-  //bool mBoolValueHolder;
-  bool* mVariable_o = nullptr;
-
-  uint8_t* mVariable_ub = nullptr;
-  uint16_t* mVariable_us = nullptr;
-  uint32_t* mVariable_ui = nullptr;
-  uint64_t* mVariable_ul = nullptr;
-  int8_t* mVariable_b = nullptr;
-  int16_t* mVariable_s = nullptr;
-  int32_t* mVariable_i = nullptr;
-  int64_t* mVariable_l = nullptr;
-  float* mVariable_f = nullptr;
-  double* mVariable_d = nullptr;
-
-  // initialize a branch
-  bool initBranch(TTree* tree);
-
-  // initialize chunk ib
-  bool initDataBuffer(Int_t ib);
+ protected:
+  std::string mBranchName;
 
- public:
-  BranchIterator(TTree* tree, std::shared_ptr<arrow::ChunkedArray> col, std::shared_ptr<arrow::Field> field);
-  ~BranchIterator();
+  virtual void resetBuffer() = 0;
+  virtual void nextChunk() = 0;
 
-  // has the iterator been properly initialized
-  bool getStatus();
+  arrow::ChunkedArray* mColumn = nullptr;
 
-  // fills buffer with next value
-  // returns false if end of buffer reached
-  bool push();
+  int64_t const* mCurrentPos = nullptr;
+  mutable int mFirstIndex = 0;
+  mutable int mCurrentChunk = 0;
+
+  int listSize = 1;
 };
 
-class TableToTree
+template <typename T>
+struct ROOTTypeString {
+  static constexpr char const* str = "/E";
+};
+
+template <>
+struct ROOTTypeString<bool> {
+  static constexpr char const* str = "/O";
+};
+
+template <>
+struct ROOTTypeString<uint8_t> {
+  static constexpr char const* str = "/b";
+};
+
+template <>
+struct ROOTTypeString<uint16_t> {
+  static constexpr char const* str = "/s";
+};
+
+template <>
+struct ROOTTypeString<uint32_t> {
+  static constexpr char const* str = "/i";
+};
+
```

Comment:
How about something like:

```
#include <iostream>


struct ConvertMe {
template <typename T> constexpr static char foo[3] = "**";
template <> constexpr static char foo<double>[3] = "do";
template <> constexpr static char foo<float>[3] = "fl";
};

int main() {
    std::cout << ConvertMe::foo<bool> << std::endl;
}
```
?

Reply:
Implemented.

Reply:
I had to revert this, because GCC 10 fails to compile it...